docs/src/modules/components/MarkdownDocs.js

```diff
@@ -12,16 +19,17 @@ import BrandingProvider from 'docs/src/BrandingProvider';
 import Ad from 'docs/src/modules/components/Ad';
 import AdGuest from 'docs/src/modules/components/AdGuest';
 
-function JoyModeObserver({ mode }) {
+function ModeObserver({ mode, useColorScheme }) {
   const { setMode } = useColorScheme();
   React.useEffect(() => {
```

Comment:
A layout effect would make more sense. See the current bug:

```suggestion
  useEnhancedEffect(() => {
```

https://github.com/mui/material-ui/assets/3165635/09261cd9-8e29-4705-9183-7c708eef644f

---

But the pattern used feels wrong in the first place. I would expect that when we change the mode, to change the mode instantly, so `useColorScheme` shouldn't be called here, but where we set the dark/light mode. Could we try that instead?